### DISCUSSION TITLE
Add cache.before init event hook or getCacheKey() filter

### DISCUSSION THREAD

**Discussion by benosman** [directus/directus#16587] [Feature Requests]
I propose adding a cache.before init event hook just before the cache middleware is loaded.

My use case is to be able to add custom query keys to `req.sanitizedQuery` for later use in my extensions, but still have those added to the cache key.

Alternatively a filter to modify the cache key directly could work.

I would be happy to create a pull request if this is to be considered.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
